sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -316,25 +319,21 @@ case class CoalescedBoundary(startReducerIndex: Int, endReducerIndex: Int)
  * fewer number of partitions.
  */
 case class CoalescedHashPartitioning(from: HashPartitioning, partitions: Seq[CoalescedBoundary])
```

Comment:
shall we simply keep `expressions: Seq[Expression]` instead of `HashPartitioning`?

Reply:
The origin HashPartitioning is mainly used to create original HashShuffleSpec to check if HashShuffleSpecs are compatiable.